### DISCUSSION TITLE
Good vendor for hosting Stashapp - any recommendations?

### DISCUSSION THREAD

**Discussion by Alextocode** [stashapp/stash#4987] [Q&A]
Hey there pervs and porn lovers :D

I would like to collect some opinions from the community.

### Current setup
Currently I am hosting my Stash on an internal VM hosted on a Hyper-V mini server I built from an Intel Nuc (and yes this require some hacking to make it working - anyway..).

### Pitfall
Long story short - the storage disk of my Nuc only has 2TB of disk space and 1TB of these I could allocate to my Stash VM but I am pretty sure this will not last on long-term. Since I did not yet get many of the porn series and other stuff I'd like to have I am in need of much more disk space. I think once I got my Stashapp populated as I am thinking to do it **it will be around 4TB**. Maybe even more. because currently I mostly added some older series with low quality and file size mostly <100MB. Still I already have 300GB filled.

### The idea
The consideration would be to **host Stash in the cloud** instead and protect it via Cloudflare with Client Certificates.
Naturally the requirements for the vendor are that they offer servers **with huge disks, but still not accordingly much memory and CPU power** I won't need. Some vendors call this "Storage VPS".

I already found this offer from Swiss server hoster Marbell => https://de.marbell.com/storage-vps-hosting.
They offer a VPS with 4TB HDD and SSD Cache and decent CPU and Memory for 15€ a month what sounds cheap to me. However I saw that their reputation in the web is not the best. So I wonder ** if someone of you guys could recommend some other good vendors**.
I like to spend around 15€ a months but not more than 20€. The cheaper the better - of course I still would like to have some good quality. I am not in need to use dumping services with poor quality just for saving a buck or two.
Ideally the hosting happens in Europe for better latency. Of course for data protection reasons Switzerland would be perfect ;)

Any smart advices are warmly welcome. 

### Other possible considerations?!
Of course I could also consider to **replace my server disk with a bigger one, but it is a SSD and SSDs in 2.5 format with 4+ TB are not very cheap** to be honest..not sure if it is worth to buy such an expensive upgrade for a 900€ Nuc Kit (including all additional components) I bought in 2019. 

**I also have a Synology NAS, but it is quite old and only has 1GB of RAM**..... Buying a new NAS could be an option but I would like to have something **that does not need much power**. My Nuc is running 24/7 an is still very silent and thrifty. Electricity rates in "great" Germany are pretty high, as you may have heart so I don't want to use devices that will spend electricity for 200€+ per year. And considering that I own a lifetime 10TB pCloud Space I am not sure if another NAS will be really a benefit - but I am still open for all suggestions.

Looking to hear from you!

Cheers and Thanks!

Alex

**Comment by feederbox826** [comment]
For pure storage, https://www.hetzner.com/storage/storage-box/ and https://contabo.com/en/storage-vps/ have been good to me so far

but honestly I would recommend just getting a larger HDD and attaching it to your NUC via USB or externally.
